hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java

```diff
@@ -140,6 +148,29 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
         fileStore.put(fileBuilder.build());
     }
 
+    private void feeCalculation(HandleContext handleContext, FileAppendTransactionBody fileAppend, File file) {
+        final var dataLength =
+                (fileAppend.contents() != null) ? fileAppend.contents().length() : 0;
+
+        final long effectiveLifeTime;
+        if (fileNumbers.isSoftwareUpdateFile(file.fileId().fileNum())) {
+            effectiveLifeTime = THREE_MONTHS_IN_SECONDS;
+        } else {
+            final var effCreationTime =
+                    handleContext.body().transactionID().transactionValidStart().seconds();
+            final var effExpiration = (file.expirationSecond() > 0) ? file.expirationSecond() : effCreationTime;
+            effectiveLifeTime = effExpiration - effCreationTime;
```

Comment:
This produces an effective lifetime of `0` when expiration is not set.  Is that the current behavior?  It seems odd that the default lifetime for fee calculation would be 0.

Reply:
@Neeharika-Sompalli  I think that what we saw with you.
That what I also see in methods metaForFileAppend

Reply:
Looks like we need to leave as-is for now, but please ensure we revisit this item in the future (after the modularization parity testing finishes).

Reply:
I added a comment for it